src/transformers/generation/configuration_utils.py

```diff
@@ -224,19 +225,20 @@ class GenerationConfig(PushToHubMixin):
         low_memory (`bool`, *optional*):
             Switch to sequential beam search and sequential topk for contrastive search to reduce peak memory.
             Used with beam search and contrastive search.
-        watermarking_args (`Dict`, *optional*):
+        watermarking_config (Union[`WatermarkingConfig`, `dict`], *optional*):
             Arguments used to watermark the model outputs by adding a small bias to randomly selected set of "green" tokens.
```

Comment:
I would add that if `dict` is passed, it will be converted to `WatermarkingConfig`